### PR TITLE
login: bring back the msi model for creds management in the cloud shell

### PR DESCRIPTION
**Pull request by yugangw-msft**
Pretty much revert #5772 which reverted the right behavior as ansible cli was not able to integrate with cloud shell at the same time. Hope this time we can get it in

//CC: @jluk @yangl900 

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5880
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py [hunk 1]
```diff
@@ -97,8 +95,8 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
           identity=False, identity_port=None,
           msi=False, msi_port=None):  # will remove msi_xxx in a future release
     """Log in to access Azure subscriptions"""
-    import os
-    import re
+    # import os
+    # import re
```
Comment (troydai) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: Just remove?

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py [hunk 2]
```diff
@@ -111,22 +109,22 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
...
+
+    # if in_cloud_console():
+    #     console_tokens = os.environ.get('AZURE_CONSOLE_TOKENS', None)
+    #     if console_tokens:
+    #         return profile.find_subscriptions_in_cloud_console_thru_raw_token(re.split(';|,', console_tokens))
+    #     logger.warning(_CLOUD_CONSOLE_WARNING_TEMPLATE, 'login')
+    #     return
+
+    # if identity or msi:
+    #     return profile.find_subscriptions_in_vm_with_msi(identity_port or msi_port or 50342, username)
```
Comment (troydai) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: Just remove?

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py [hunk 3]
```diff
@@ -165,11 +163,11 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
...
     """Log out to remove access to Azure subscriptions"""
-    # if in_cloud_console():
-    #     logger.warning(_CLOUD_CONSOLE_LOGOUT_WARNING)
     if in_cloud_console():
-        logger.warning(_CLOUD_CONSOLE_WARNING_TEMPLATE, 'logout')
-        return
+        logger.warning(_CLOUD_CONSOLE_LOGOUT_WARNING)
+    # if in_cloud_console():
+    #     logger.warning(_CLOUD_CONSOLE_WARNING_TEMPLATE, 'logout')
+    #     return
```
Comment (troydai) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: Remove these lines?

**Comment by yugangw-msft** [comment]
> Do we still need the commented code?
 > You may need to modify HISTORY.rst

Done

**Comment by yugangw-msft** [comment]
Mark as "do not merge" as Cloud Shell team is still verifying it

**Pull request merged**